libs/ui-lib/lib/ocm/components/clusterDetail/utils.tsx

```diff
@@ -98,3 +97,21 @@ export const getMostSevereHostStatus = (hosts: Host[]) => {
   }
   return status;
 };
+
+const download = (fileUrl?: string, dataBlob?: Blob, fileName?: string) => {
+  const link = document.createElement('a');
+  if (fileUrl && fileUrl !== '') {
+    link.setAttribute('href', fileUrl);
+  }
+  if (dataBlob) {
+    const file = new Blob([dataBlob], { type: 'text' });
+    link.setAttribute('href', URL.createObjectURL(file));
+  }
+  if (fileName) {
+    link.setAttribute('download', fileName);
+  }
+  link.style.visibility = 'hidden';
+  document.body.appendChild(link);
+  link.click();
+  document.body.removeChild(link);
+};
```

Comment:
Just as a suggestion...

```suggestion
export const download = (fileUrlOrDataBlob: string | Blob, fileName?: string) => {
  const link = document.createElement('a');
  link.style.visibility = 'hidden';
  if (fileName) {
    link.setAttribute('download', fileName);
  }
  
  if (typeof fileUrlOrDataBlob === 'string') {
    link.setAttribute('href', fileUrl);
  } else if (somehowCheckIsBlob(fileUrlOrDataBlob)) {
    const file = new Blob([dataBlob], { type: 'text' });
    link.setAttribute('href', URL.createObjectURL(file));
  } else {
    // handle invalid argument exception ;-P
  }
  
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
};
```

Reply:
Good suggestion, but at the moment I'll merge like that.